services/rpcfilters/transaction_sent_to_upstream_event_test.go

```diff
@@ -0,0 +1,81 @@
+package rpcfilters
+
+import (
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/assert"
+)
+
+var transactionHashes = []common.Hash{common.HexToHash("0xAA"), common.HexToHash("0xBB"), common.HexToHash("0xCC")}
+
+func TestTransactionSentToUpstreamEventSubscribe(t *testing.T) {
+	event := newTransactionSentToUpstreamEvent()
+	assert.NoError(t, event.Start())
```

Comment:
You may consider using `github.com/stretchr/testify/require` in some places. For example here, doing `assert.NoError()` won't stop execution immediately, however, if `event.Start()` fails, there is no point to check other asserts in this function.